### PR TITLE
Lightning qubit2 upgrade api

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This PR merges #607 and #627 to enable integration testing.

**Description of the Change:**
Since #607 enables QNode execution, several tests under `tests` (namely all except those explicitly testing the old device API) become available, in principle that is since features like adjoint at still missing. This PR therefore introduces (as a temporary measure and until all old device API backends become deprecated) a `_new_API` attribute which allows discriminating between backends. All tests with missing functionality (e.g. adjoint, shots, etc.) are marked `skip` or `xfail`. 

**Benefits:**
Leverage unit/integration tests that already exists.
Enable coexistence of old/new device API backends, likely needed if we cannot complete the port of L-GPU/Kokkos before next release.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning/lightning_qubit/lightning_qubit2.py [hunk 1]
```diff
@@ -194,7 +201,13 @@ class LightningQubit2(Device):
 
     _device_options = ("rng", "c_dtype", "batch_obs", "mcmc", "kernel_name", "num_burnin")
 
+    _new_API = True
     _CPP_BINARY_AVAILABLE = LQ_CPP_BINARY_AVAILABLE
+    short_name = "lightning.qubit2"
+    operations = _operations
+    observables = _observables
```
Comment (albi3ro) [inline pennylane_lightning/lightning_qubit/lightning_qubit2.py:208]: So we do have these as class properties below as well?

File: pennylane_lightning/lightning_qubit/lightning_qubit2.py [hunk 2]
```diff
@@ -194,7 +201,13 @@ class LightningQubit2(Device):
 
     _device_options = ("rng", "c_dtype", "batch_obs", "mcmc", "kernel_name", "num_burnin")
 
+    _new_API = True
     _CPP_BINARY_AVAILABLE = LQ_CPP_BINARY_AVAILABLE
+    short_name = "lightning.qubit2"
```
Comment (albi3ro) [inline pennylane_lightning/lightning_qubit/lightning_qubit2.py:206]: Was the lack of a `short_name` breaking something in particular?

Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/lightning_qubit2.py:206]: At first I could not properly initialize a device with it's name and I thought I should add this by analogy with `lightning.qubit`. But the issue was the missing symlink `lightning_qubit2`, I think. So we can remove or keep it.

File: pennylane_lightning/lightning_qubit/lightning_qubit2.py (hunk 1 above)
Comment (vincentmr) [reply pennylane_lightning/lightning_qubit/lightning_qubit2.py:208]: I think the duplication was introduced during a merge. Let me keep only these two.

**Pull request merged**